redis/reply.go

```diff
@@ -477,3 +477,44 @@ func Positions(result interface{}, err error) ([]*[2]float64, error) {
 	}
 	return positions, nil
 }
+
+func Uint64s(reply interface{}, err error) ([]uint64, error) {
+	var result []uint64
+	err = sliceHelper(reply, err, "Uint64s", func(n int) { result = make([]uint64, n) }, func(i int, v interface{}) error {
+		switch v := v.(type) {
+		case uint64:
+			result[i] = v
+			return nil
+		case []byte:
+			n, err := strconv.ParseUint(string(v), 10, 64)
+			result[i] = n
+			return err
+		default:
+			return fmt.Errorf("redigo: unexpected element type for Uint64s, got type %T", v)
+		}
+	})
+	return result, err
+}
+
+func Uint64Map(result interface{}, err error) (map[uint64]uint64, error) {
+	values, err := Values(result, err)
+	if err != nil {
+		return nil, err
+	}
+	if len(values)%2 != 0 {
+		return nil, errors.New("redigo: Uint64Map expects even number of values result")
+	}
+	m := make(map[uint64]uint64, len(values)/2)
+	for i := 0; i < len(values); i += 2 {
+		key, err := Uint64(values[i], nil)
+		if err != nil {
+			return nil, errors.New("redigo: Uint64Map key not a bulk uint4 value")
```

Comment:
```suggestion
			return nil, err
```
I know that other areas of the code use this style, but I think its better to not loose the original error which includes more detail, thoughts?

If not I believe the `bulk uint4` should actually be `bulk string`